### PR TITLE
Replace 30 some if statements with switch statements

### PR DESCRIPTION
**Pull request by DanVanAtta**
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/795)

### CODE REVIEW

**Comment by RoiEXLab** [comment]
@DanVanAtta 2 errors on line 215 in GameDataExporter.java:
propName is already defined, and propertyNames does not exist...

File: src/games/strategy/engine/data/GameParser.java [hunk 1]
```diff
@@ -592,28 +592,37 @@ protected static void setGrids(final GameData data, final String gridType, final
...
-      throw new GameParseException("vertical-connections attribute must be either \"explicit\" or \"implicit\"");
+    switch (verticalConnections) {
+      case "implicit":
+        verticalConnectionsImplict = true;
+        break;
+      case "explicit":
+        verticalConnectionsImplict = false;
+        break;
+      default:
+        throw new GameParseException("vertical-connections attribute must be either \"explicit\" or \"implicit\"");
```
Comment (ron-murhammer) [inline src/games/strategy/engine/data/GameParser.java]: Seems like some duplication here with a good opportunity for a helper method.

File: src/games/strategy/engine/data/GameParser.java [hunk 2]
```diff
@@ -931,22 +940,27 @@ private void parseProperties(final Node root) throws GameParseException {
...
               }
-            }
-            properties.set(property, intValue);
-          } else {
-            properties.set(property, value);
+              properties.set(property, intValue);
+              break;
+            default:
+              properties.set(property, value);
+              break;
```
Comment (ron-murhammer) [inline src/games/strategy/engine/data/GameParser.java]: Is a 'break' needed here?

File: src/games/strategy/engine/data/GameParser.java [hunk 3]
```diff
@@ -984,30 +998,40 @@ private void parseEditableProperty(final Element property, final String name, fi
...
-    if (childName.equals("boolean")) {
-      editableProperty = new BooleanProperty(name, null, Boolean.valueOf(defaultValue).booleanValue());
-    } else if (childName.equals("file")) {
-      editableProperty = new FileProperty(name, null, defaultValue);
-    } else if (childName.equals("list") || childName.equals("combo")) {
-      final StringTokenizer tokenizer = new StringTokenizer(child.getAttribute("values"), ",");
-      final Collection<String> values = new ArrayList<>();
-      while (tokenizer.hasMoreElements()) {
-        values.add(tokenizer.nextToken());
+    switch (childName) {
```
Comment (ron-murhammer) [inline src/games/strategy/engine/data/GameParser.java]: Not sure this is actually easier to read/maintain converting to a switch. Once you have logic in checks where multiple cases map to the same logic, its probably easier to read as if/else.

File: src/games/strategy/engine/data/export/GameDataExporter.java [hunk 4]
```diff
@@ -212,14 +212,19 @@ private void printEditableProperty(final IEditableProperty prop) {
...
     for(String propName : conProperties.keySet()) {
-      if (propName.equals("notes")) { // TODO: unchecked reflection
-        // Special handling of notes property
-        printNotes((String) conProperties.get(propName));
-      } else if (propName.equals("EditMode") || propName.equals("GAME_UUID")
-          || propName.equals("games.strategy.engine.framework.ServerGame.GameHasBeenSaved")) {
-        // Don't print these options
-      } else {
-        printConstantProperty(propName, conProperties.get(propName));
+      switch (propName) {
```
Comment (ron-murhammer) [inline src/games/strategy/engine/data/export/GameDataExporter.java]: Not sure this is actually easier to read/maintain converting to a switch. Once you have logic in checks where multiple cases map to the same logic, its probably easier to read as if/else.

File: src/games/strategy/engine/lobby/client/ui/LobbyFrame.java [hunk 5]
```diff
@@ -273,18 +280,25 @@ private void showServerMessage(final LobbyServerProperties props) {
...
-        ticks = result2 * 1000 * 60 * 60;
-      } else if (selectedTimeUnit.equals("Day")) {
-        ticks = result2 * 1000 * 60 * 60 * 24;
-      } else if (selectedTimeUnit.equals("Week")) {
-        ticks = result2 * 1000 * 60 * 60 * 24 * 7;
-      } else if (selectedTimeUnit.equals("Month")) {
-        ticks = result2 * 1000 * 60 * 60 * 24 * 30;
-      } else if (selectedTimeUnit.equals("Year")) {
-        ticks = result2 * 1000 * 60 * 60 * 24 * 365;
+      switch (selectedTimeUnit) {
```
Comment (ron-murhammer) [inline src/games/strategy/engine/lobby/client/ui/LobbyFrame.java]: Seems like some duplication here with a good opportunity for a helper method.

File: src/games/strategy/engine/lobby/server/headless/HeadlessLobbyConsole.java [hunk 6]
```diff
@@ -85,23 +85,32 @@ private void process(final String command) {
...
+      case "memory":
+        memory();
+        break;
+      case "threads":
+        threads();
+        break;
+      default:
+        out.println("unrecognized command:" + command);
+        showHelp();
+        break;
```
Comment (ron-murhammer) [inline src/games/strategy/engine/lobby/server/headless/HeadlessLobbyConsole.java]: Is the 'break' needed here?

File: src/games/strategy/engine/random/ScriptedRandomSource.java [hunk 7]
```diff
@@ -40,12 +40,16 @@ public ScriptedRandomSource() {
...
+      switch (token) {
+        case "e":
+          m_numbers[i] = ERROR;
+          break;
+        case "p":
+          m_numbers[i] = PAUSE;
+          break;
+        default:
+          m_numbers[i] = Integer.parseInt(token) - 1;
+          break;
```
Comment (ron-murhammer) [inline src/games/strategy/engine/random/ScriptedRandomSource.java]: Is the 'break' needed here?

**Comment by ron-murhammer** [comment]
@DanVanAtta I reviewed the first half or so of this PR and had some feedback that seems like it going to continue to repeat for the rest.

**Comment by DanVanAtta** [comment]
Yeah, I'm very meh on this.. looks like this type of thing needs to be done by hand to be effective, only a couple of these updates are nice. Closing, bigger fish to fry, and looks like we'll have to fix this in vertical slices, rather than a cross the board, horizontal operation

**Pull request closed without merging**